### PR TITLE
HDDS-1628. Fix the execution and return code of smoketest executor shell script

### PR DESCRIPTION
**Pull request by elek**
Problem: Some of the smoketest executions were reported to green even if they contained failed tests.

Root cause: the legacy test executor (hadoop-ozone/dist/src/main/smoketest/test.sh) which just calls the new executor script (hadoop-ozone/dist/src/main/compose/test-all.sh) didn't handle the return code well (the failure of the smoketests should be signalled by the bash return code)

This patch:
 * Fixes the error code handling in smoketest/test.sh
 * Fixes the test execution in compose/test-all.sh (should work from any other directories)
 * Updates hadoop-ozone/dev-support/checks/acceptance.sh to use the newer test-all.sh executor instead of the old one.

See: https://issues.apache.org/jira/browse/HDDS-1628

### CODE REVIEW

File: hadoop-ozone/dev-support/checks/acceptance.sh [hunk 1]
```diff
@@ -13,6 +13,7 @@
 # WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 # See the License for the specific language governing permissions and
 # limitations under the License.
+DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
 export HADOOP_VERSION=3
-hadoop-ozone/dist/target/ozone-*-SNAPSHOT/smoketest/test.sh
+$DIR/../../../hadoop-ozone/dist/target/ozone-*-SNAPSHOT/compose/test-all.sh
```
Comment (hadoop-yetus) [inline hadoop-ozone/dev-support/checks/acceptance.sh]: shellcheck:1: note: Double quote to prevent globbing and word splitting. [SC2086]

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 1390 | Docker mode activated. |
||| _ Prechecks _ |
| +1 | dupname | 0 | No case conflicting files found. |
| 0 | shelldocs | 0 | Shelldocs was not available. |
| +1 | @author | 0 | The patch does not contain any @author tags. |
| -1 | test4tests | 0 | The patch doesn't appear to include any new or modified tests.  Please justify why no new tests are needed for this patch. Also please list what manual steps were performed to verify this patch. |
||| _ trunk Compile Tests _ |
| 0 | mvndep | 135 | Maven dependency ordering for branch |
| +1 | mvninstall | 727 | trunk passed |
| +1 | mvnsite | 0 | trunk passed |
| +1 | shadedclient | 1115 | branch has no errors when building and testing our client artifacts. |
||| _ Patch Compile Tests _ |
| 0 | mvndep | 34 | Maven dependency ordering for patch |
| +1 | mvninstall | 617 | the patch passed |
| +1 | mvnsite | 0 | the patch passed |
| -1 | shellcheck | 0 | The patch generated 1 new + 1 unchanged - 0 fixed = 2 total (was 1) |
| +1 | whitespace | 0 | The patch has no whitespace issues. |
| +1 | shadedclient | 852 | patch has no errors when building and testing our client artifacts. |
||| _ Other Tests _ |
| +1 | unit | 121 | hadoop-hdds in the patch passed. |
| +1 | unit | 254 | hadoop-ozone in the patch passed. |
| +1 | asflicense | 70 | The patch does not generate ASF License warnings. |
| | | 5588 | |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=18.09.5 Server=18.09.5 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-902/1/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/902 |
| Optional Tests | dupname asflicense mvnsite unit shellcheck shelldocs |
| uname | Linux fdec2b2a02dd 4.15.0-48-generic #51-Ubuntu SMP Wed Apr 3 08:28:49 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux |
| Build tool | maven |
| Personality | personality/hadoop.sh |
| git revision | trunk / 7991159 |
| shellcheck | https://builds.apache.org/job/hadoop-multibranch/job/PR-902/1/artifact/out/diff-patch-shellcheck.txt |
|  Test Results | https://builds.apache.org/job/hadoop-multibranch/job/PR-902/1/testReport/ |
| Max. process+thread count | 294 (vs. ulimit of 5500) |
| modules | C: hadoop-ozone hadoop-ozone/dist U: hadoop-ozone |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-902/1/console |
| versions | git=2.7.4 maven=3.3.9 shellcheck=0.4.6 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

**Comment by elek** [comment]
Thanks the review @xiaoyuyao 

I am merging it to the trunk (shellcheck is fixed)

**Comment by hadoop-yetus** [comment]
:broken_heart: **-1 overall**






| Vote | Subsystem | Runtime | Comment |
|:----:|----------:|--------:|:--------|
| 0 | reexec | 128 | Docker mode activated. |
| -1 | patch | 14 | https://github.com/apache/hadoop/pull/902 does not apply to trunk. Rebase required? Wrong Branch? See https://wiki.apache.org/hadoop/HowToContribute for help. |


| Subsystem | Report/Notes |
|----------:|:-------------|
| Docker | Client=18.09.5 Server=18.09.5 base: https://builds.apache.org/job/hadoop-multibranch/job/PR-902/2/artifact/out/Dockerfile |
| GITHUB PR | https://github.com/apache/hadoop/pull/902 |
| Console output | https://builds.apache.org/job/hadoop-multibranch/job/PR-902/2/console |
| versions | git=2.7.4 |
| Powered by | Apache Yetus 0.10.0 http://yetus.apache.org |


This message was automatically generated.

**Pull request closed without merging**